src/transformStateWithClones.js

```diff
@@ -8,6 +8,37 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  let stateCopy = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        stateCopy = { ...stateCopy };
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        stateCopy = { ...stateCopy };
+
+        for (const key of action.keysToRemove) {
+          delete stateCopy[key];
+        }
+        break;
+
+      case 'clear':
+        stateCopy = { ...stateCopy };
+
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+        break;
+    }
```

Comment:
switch requires a `default:` case as a fallback option for cases not covered
usually it's `throw new Error('error explanation')`

Reply:
ill keep in mind, thank you